### ISSUE TITLE
Confused about services

### ISSUE THREAD

**Issue by vitobotta** [presslabs/mysql-operator#313]
For starters, thanks a lot for this operator! I especially like the built in backup/restore feature.

I am a bit confused as to which service to use when. I see three services:

- mysql-cluster-mysql
- mysql-cluster-mysql-master
- mysql-cluster-mysql-nodes

What should I use in my applications? Does mysql-cluster-mysql do automatic splitting reads/writes between master and slaves?

Thanks in advance for the clarification.

**Comment by AMecea** [comment]
Hi @vitobotta, Indeed the services are a little confusing. More details about this can be found on the documentation [page](https://www.presslabs.com/code/mysqloperator/integrate-operator/) of the operator.

To summarise it up:
 * `mysql-cluster-mysql-master` is the service that always points to the master node, and this should be used in your application.
 * `mysql-cluster-mysql` is a service that points to all healthy MySQL nodes(that replicates, and are not lagged, including the master node). If your application allows you to configure a split database connections then for reading use this service and for writing use the master service. This service should NOT be used for writes. Does not splits the reads and writes.
 * the last service `mysql-cluster-mysql-nodes` is a headless service used internally for communication between nodes. In the next version (`v0.3`) this service will be removed and replaced by a global one.

We plan to include ProxySQL but we delayed it because we think that ProxySQL should be close to the application instead to the database server.

Hope this helps,
Best regards

**Comment by vitobotta** [comment]
Hi @AMecea, thanks for the clarification! I guess I'll have to figure out how to split reads and writes. Thanks again.

**Comment by vitobotta** [comment]
Hi! I am finally looking to implement read/write splitting in my app after a long delay... since Rails 6 has a built in support for this stuff now so it should be easier. With a cluster created with the current version of the operator I have still three different services:

1. mysql
2. mysql-cluster-mysql
3. mysql-cluster-mysql-master

So the master service is still the same and I know I should use that for the writes. But for the reads should I use 1 or 2? I would guess it's still 2 since the name is the same and it has a selector for healthy nodes, but I wanted to ask just to be sure. Is `mysql` just the new name for `mysql-cluster-mysql-nodes`? Thanks!

**Comment by AMecea** [comment]
Yes, `mysql` is in a way a rename of `mysql-cluster-mysql-nodes` but has a selector for all nodes from all clusters.

It's a shared service. We choose to use a short name for this service to have shorter hostnames (i.e. `mysql-cluster-mysql-0.mysql.default`). The reason behind this decision is a MySQL limitations, see https://github.com/presslabs/mysql-operator/issues/170.
